src/libraries/System.Runtime/tests/LinkerTests/GenericArraySortHelperTest.cs

```diff
@@ -0,0 +1,16 @@
+using System;
+using System.Collections;
+
+class Program
+{
+    static int Main(string[] args)
+    {
+        var x = new[] { 5, 4, 3, 100, 2, 1 };
+        var y = new[] { "a", "b", "c", "h", "d", "e" };
+        // This will test that GenericArraySortHelper'2 called by reflection will be kept.
+        Array.Sort(x, y);
```

Comment:
Do we want a simple verification that the arrays are in fact sorted?

Reply:
For this particular test, if GenericArraySortHelper is trimmed out, this Array.Sort call will throw an exception at runtime when trying to instantiate it, so no need to check if the Array was actually sorted as we really want minimal code so other things don't cause GenericArraySortHelper to get rooted. I have tried undoing my changes to remove the attribute annotation and verified that this test would fail in that case and it did throw the exception as expected and failed the test.